### DISCUSSION TITLE
Simulation of very short monopole with series inductor and parallel capacitor matching

### DISCUSSION THREAD

**Discussion by kbocheltree** [thliebig/openEMS-Project#131] [Q&A Usage]
I would like to simulate a very short monopole (approximately 1/60th a wavelength) operating at 150 MHz with a series inductor (with known resistance) and parallel capacitor that provides matching to give a 50 ohm input impedance at 150 MHz. I know the approximate values of the required matching from measurements on short monopoles. I would like to see both the farfield radiation pattern and the input impedance vs frequency.

Is openEMS suited to simulations where the antenna is very small compared to the wavelength?  Are there any special considerations to the meshing for such configurations? 

Is RLC support in the latest openEMS builds so that these matching elements can be simulated?

**Comment by Sarajevo67** [comment]
RLC is not yet supported, but PR is waiting to be merged. Please take a look at this thread: #65

**Reply by kbocheltree** [reply]
[Sarajevo67](https://github.com/Sarajevo67), thanks for the pointer. So it looks like when https://github.com/thliebig/openEMS-Project/discussions/65 is merged, then Inductors will be support. Is there a scheduled time when https://github.com/thliebig/openEMS-Project/discussions/65 will be merged?

**Comment by gadiLhv** [comment]
**Matching network**
In the meanwhile, it is possible to simulate a 2nd port (or 3 more ports, for a proper pi\T network), generate an SnP file (with sequential excitation) and add the components using a 3rd party software. 
At your convenience, please view #100.

**Mesh considerations**
This is a tricky one. You want the mesh to be fine enough to conform to your (small) model, but not too fine, for the simulation to end in your lifetime.

**Airbox** 
You didn't ask, but I will answer. Airboxes in low frequencies are a bit of an issue. If your small Antenna just so happens to be efficient, you might need a rather large airbox. This requires proper mesh grading, for starters, and may increase the simulation domain significantly (memory issues, longer sim, etc.).
There is no gospel here. Just saying that this may require some experimemtation.

**Reply by VolkerMuehlhaus** [reply]
I would not even try to include the matching elements in EM simulation here. 

I use FDTD EM simulations a lot for antenna design, and usually separate the antenna simulation part (EM) from matching circuit simulation (circuit design). If you design a proper layout with low parasitics, matching can be designed very well without using EM, especially at your low frequency of only 150 MHz.

**Reply by gadiLhv** [reply]
Hey @VolkerMuehlhaus.
This is essentially what I offered. Separate the antenna design from the matching network, while adding the components in a circuit simulator, rather than EM simulation.

Do you have any other remarks on the two other points.

**Reply by VolkerMuehlhaus** [reply]
No, I fully agree on your post!

With my post I just wanted to emphasize that there's no need to include the lumped elements into the EM part, and it's actually more efficient to run this externally in circuit simulation without re-simulating the entire FDTD problem for every change of lumped element values. 

The FDTD solver that I use in my job (EM support and consulting) does support RLC elements inside the FDTD model, but for the reason stated above I prefer to use ports for the discrete elements (if necessary at all, you could possibly simulate the antenna alone and consider the routing of antenna matching network ideal at 150 Mhz if you are familar with good RF routing practices) and then tweak the antenna matching network in circuit simulation (e.g. QUCS)

The other points that you mentioned are very valid and important here indeed!!

Best regards
Volker

**Reply by kbocheltree** [reply]
@gadiLhv and @VolkerMuehlhaus,
Thanks for your informative input! I am new to EM and FDTD simulation, but was wanting to simulate the matching with the antenna because the antenna is so short electrically that the impedance is on the order of 10-j600 ohms (with a VSWR of around 600) at 150 MHz without the matching network. (This short antenna is needed due to the small volume that is available for this circuit and antenna).

This presents a few issues:
1. The impedance of the antenna is so high, I was not sure how to represent the input port high impedance in the simulation
2. The matching network will have to be very precisely matched to reduce reflections. I wanted to see how the matching network and the antenna worked together across a frequency range to see the bandwidth of the matching network with the antenna.
3. I was considering trimming the antenna length to tune the matching and I wanted to confirm that the antenna with proper matching and trimming brings the VSWR close to 1.

Is there a good circuit simulator that can be coupled with openEMS and the simulation run together?

I am not yet familiar with the use of the Airboxes and the tradeoffs in using them. I have worked through some of the tutorials with airboxes, but there was not much description of the proper use of airboxes.

Thanks again!,
Ken

**Reply by VolkerMuehlhaus** [reply]
Hi Ken,

no problem, we will try to support you!

Let's adress the basics first: you can split multiple blocks (such as antenna and matching network) **IF** there is no field coupling between them, and they only interact through the circuit nodes (ports). That's true to almost any antenna & matching circuit that I ever worked with. In this case, you can simulate both parts separately, and cascade the S-parameter blocks in circuit simulation (e.g. QUCS or [QUCS Studio](http://qucsstudio.de/))

For the antenna, you want to use EM simulation for sure, but the matching part can be done in (fast!) circuit simulation. The reason to split this is simple; if you include the matching elements in EM simulation, you need to repeat the full field simulation over and over again for each change of LC values, although these elements have no impact on field distribution. They only have an effect on input impedance, but that is something we can easily handle by node-based circuit simulation (cascade antenna block and matching circuit block). This way, you run the EM simulation of the antenna only once, and the rest is simple circuit simulation (node/port based).

For your antenna model, you can use a **standard 50 Ohm port impedance**, like you would do in measurement. Of course, your "raw" antenna will not be matched to 50 Ohm, but the resulting S-parameters will give an accurate description of the antenna impedance over frequency. 

You didn't mention what environment you want to use for OpenEMS, Matlab or Python? 
I can post a code snippet to create *.s1p output for Python, if that is what you need.

One important aspect here is properly include the **loss in LC matching elements**, that will have a large impact on antenna efficiency for such a short radiator with rather extreme impedance matching ratio. By using circuit simulation instead of embedding LC in the FDTD model, it is easy to use "real world" component data that are far more accurate than ideal LC.

You wrote above that you want to tweak the **antenna length** - in that case you would need one EM model (S-parameter set) for each length. Or you could use a fixed length and tweak the LC matching elements.

What is very important in any case for short radiators is the **ground plane**, this has a large effect on input impedance. So you should be accurate on modelling the correct ground plane dimensions in your EM Model.

Regarding the **airbox**, that just refers to the distance between your drawn structure and the simulation boundary (absorbing boundary PML). Distance of 1/4 wavelength on all sides will be fine.

Good luck!
Volker

**Reply by kbocheltree** [reply]
@VolkerMuehlhaus,
I have been doing my circuit design in KIcad and circuit modelling with ngspice. I could use QUCS if that would be advantageous for working with the S-parameter blocks. 

So it sounds like the best approach is to run the EM simulation to get an S-parameter model of the antenna and then to run that in QUCS with the S-parameter model of the matching. I do have resistance values for the inductors I will be using for matching, so I can add that to the model.

So far, I have been primarily using the Octave version of OpenEMS, because there are more documented tutorials in that format, but I am am equally comfortable with Python. Yes an example of a code snippet in Python to create an*.s1p output would be helpful.

The ground plane is a bit difficult to determine as this circuit is in a dart body that has a small ground region of a cm in length, but the dart is then embedded in a deer for tracking, where the deer would act as a lossy ground plane. Adding a representation of this larger ground plane to my simulation should give more accurate results.

I was also considering sending out some quick turn instances of the antenna with different dimensions to have some physical instances for confirming results.

Vielen Dank!
Ken

**Reply by VolkerMuehlhaus** [reply]
Hi Ken,

I have been using ngspice for an OSS project, and know that it can do S-parameter simulation and write the output S-parameters in Touchstone format. However, I'm not sure if/how you can use circuit blocks defined by S-parameters. Coming from the world of RF simulation, I feel more familiar with tools like QUCS/QUCS Studio, so my experience with ngspice is very limited.

Ground plane: deer means that animal, right? That sounds like a truely interesting simulation task! I know that in commercial EM solvers, models involving human bodies use "phantoms" that mimic the different body structures and properties, and the high water content means really large loss indeed. That will certainly have a large impact on results.

This is the code snippet that I use for writing S1P output (Touchstone format) in Python:

 ```
   # create Touchstone S1P output file in simulation data path
    s1p_name = os.path.join(sim_path, model_basename + '.s1p')
    s1p_file = open(s1p_name, 'w')
    s1p_file.write('#   Hz   S  RI   R   50\n')
    s1p_file.write('!\n')
    for index in range(0, numfreq):
        freq = f[index]
        re = real(s11[index])
        im = imag(s11[index])
        s1p_file.write(str(freq) + ' ' + str(re) + ' ' + str(im) + '\n')
    s1p_file.close()
```
For Matlab syntax there is an openEMS function write_touchstone() in circuit toolbox (CTB) but that dind't work with 1-port data when I tried. I did a quick & dirty fix to the code to make it work for 1-port data, but don't want to share that here.

Best regards
Volker

**Reply by kbocheltree** [reply]
Hi Volker,
Thanks for the Python code snippet for writing the Touchstone S1P format. I am familiar with the Touchstone format as a VNA output. I will start coding a simple version of the antenna configuration today in Python.

Yes, this radio ends up with a ground plane through a deer or other wild animal for tracking after darting. That does make for a variable configuration lossy ground plane.

I was considering running a quick turn fabrication of 1/2 dozen different antenna and antenna/matching configurations for testing in parallel with the simulation. As I have a small volume available for the antenna, there are limited options on what can be done.

Thanks,
Ken
